src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,15 @@
+package duke;
+
+/**
+ * Signals any error sent by Duke.
+ */
+public class DukeException extends Exception {
```

Comment:
I've noticed that you have taken care of many cases of invalid input, which is really good! What you may wanna consider is to create classes for exceptions of the same type, especially because many of the exceptions you caught print the same message. These exceptions can extend `DukeException`.